### ISSUE TITLE
Loki Query Builder: No warning about losing parts of the query with a constant query

### ISSUE THREAD

**Issue by matyax** [grafana/grafana#72334]
With a query that is only a constant number, when we switch to Builder we get no widget to edit it nor message about losing this in the transition from Code to Builder.

![Image](https://github.com/grafana/grafana/assets/1069378/cf1b5ebf-3173-47dc-aae5-9bc46df90e6a)

**Comment by gtk-grafana** [comment]
Created a quick PoC fix for this: https://github.com/grafana/grafana/pull/74860

**Comment by gtk-grafana** [comment]
So after diving into the visual query editor and our parsing a bit more as part of onboarding I would argue that this is a nice to have, but not worth prioritizing this issue.
Reasoning being the only need for a scalar value is to express `1 < count_over_time({age="new"}[1m])`, as our UI in the visual query editor assumes a scalar value on the right side of the equation. So a workaround would be to rewrite the equation as `count_over_time({app="aggregator"}[$__auto]) > bool 1`.

As a note, it appears as though loki always interprets binary operations that have a scalar on the left side as if a `bool` modifier was supplied, but if the scalar is on the right side the bool modifier will actually change the output.

No wait, ignore everything above, it appears as though putting the scalar on the left side of a binary operation in loki does change the output.

Because if we try to rewrite `10 <= count_over_time({age="new"}[$__auto])`  the same way, we'd get `count_over_time({age="new"}[$__auto]) > bool 10`, which returns very different output.
<img width="1179" alt="image" src="https://github.com/grafana/grafana/assets/109082771/8af5af73-5eca-4aaa-944b-65264be6a726">

It appears to be returning the input scalar when the expression is true, but it does not have the behavior of (bool). I bet there's some code somewhere that's returning the thing on the left when the expression is true, instead of returning the vector when expression is true, this is why we see the expected calculated value when the scalar is on the right, and the input scalar is returned as the value when the scalar is on the left.

<img width="1196" alt="image" src="https://github.com/grafana/grafana/assets/109082771/c92d90d9-d162-4b60-8192-602556f41e04">
<img width="1187" alt="image" src="https://github.com/grafana/grafana/assets/109082771/30e8c894-690e-4cc6-b8b7-fab8a27302e1">
<img width="1195" alt="image" src="https://github.com/grafana/grafana/assets/109082771/bce670ab-068b-4401-a27b-d8e4a6f46d23">

**Comment by gtk-grafana** [comment]
I will follow up on the above with Logs/Loki folks tomorrow.
https://grafana.com/docs/loki/latest/query/#comparison-operators
TL;DR:
Docs do not specify that scalar/vector is different then vector/scalar

Live examples:
[scalar < vector](https://play.grafana.org/explore?panes=%7B%22fXy%22:%7B%22datasource%22:%22grafanacloud-logs%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22expr%22:%22100%20%3C%20count_over_time%28%7Bcluster%3D%5C%22play-db-cluster%5C%22%7D%5B$__auto%5D%29%22,%22queryType%22:%22range%22,%22datasource%22:%7B%22type%22:%22loki%22,%22uid%22:%22grafanacloud-logs%22%7D,%22editorMode%22:%22code%22%7D%5D,%22range%22:%7B%22from%22:%22now-6h%22,%22to%22:%22now%22%7D%7D%7D&schemaVersion=1&orgId=1)

[vector < scalar](https://play.grafana.org/explore?panes=%7B%22fXy%22:%7B%22datasource%22:%22grafanacloud-logs%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22expr%22:%22count_over_time%28%7Bcluster%3D%5C%22play-db-cluster%5C%22%7D%5B$__auto%5D%29%20%3E%20100%22,%22queryType%22:%22range%22,%22datasource%22:%7B%22type%22:%22loki%22,%22uid%22:%22grafanacloud-logs%22%7D,%22editorMode%22:%22code%22%7D%5D,%22range%22:%7B%22from%22:%22now-6h%22,%22to%22:%22now%22%7D%7D%7D&schemaVersion=1&orgId=1)
